src/actionobserver.cpp

```diff
@@ -137,44 +137,49 @@ void RunningAction::statusAccountIdChanged(const QMailAccountId &accountId)
 
 }
 
-ActionObserver::ActionObserver(QObject *parent) :
-    QObject(parent),
-    _accountsCache(new AccountsCache(this)),
-    _actionObserver(new QMailActionObserver(this))
+ActionObserver::ActionObserver(QObject *parent)
+    : QObject(parent)
+    , _accountsCache(new AccountsCache(this))
+    , _actionObserver(new QMailActionObserver(this))
 {
-    connect(_actionObserver, SIGNAL(actionsChanged(QList<QSharedPointer<QMailActionInfo> >)),
-        this, SLOT(actionsChanged(QList<QSharedPointer<QMailActionInfo> >)));
+    connect(_actionObserver, &QMailActionObserver::actionsChanged,
+            this, &ActionObserver::actionsChanged);
 }
 
 // Report only long sync type of actions.
 // Small actions like exporting updates and flags are ignored, same for actions
 // that only happen on email client UI is visible(dowload inline images, message parts, ...)
 bool ActionObserver::isNotificationAction(QMailServerRequestType requestType)
 {
-    if (requestType == TransmitMessagesRequestType || requestType == RetrieveFolderListRequestType
-            || requestType == RetrieveMessageListRequestType || requestType == RetrieveMessagesRequestType
-            || requestType == RetrieveMessageRangeRequestType || requestType == RetrieveAllRequestType
-            || requestType == SynchronizeRequestType || requestType == RetrieveNewMessagesRequestType) {
-        return true;
-    }
-    return false;
+    return requestType == TransmitMessagesRequestType
+            || requestType == RetrieveFolderListRequestType
+            || requestType == RetrieveMessageListRequestType
+            || requestType == RetrieveMessagesRequestType
+            || requestType == RetrieveMessageRangeRequestType
+            || requestType == RetrieveAllRequestType
+            || requestType == SynchronizeRequestType
+            || requestType == RetrieveNewMessagesRequestType;
 }
 
 // ################ Slots #####################
 
 void ActionObserver::actionsChanged(QList<QSharedPointer<QMailActionInfo> > actionsList)
 {
-    foreach(QSharedPointer<QMailActionInfo> action, actionsList) {
+    for (QSharedPointer<QMailActionInfo> action : actionsList) {
```

Comment:
I'm arriving a bit late, sorry. Since `actionList` is not const, this will create a copy before iterating. Is it an issue ? Or the copy on write of Qt containers will make this copy penalty neglectible ?